mmv1/products/notebooks/api.yaml

```diff
@@ -870,8 +881,7 @@ objects:
           - !ruby/object:Api::Type::Boolean
             name: "enableHealthMonitoring"
             description: |
-              Verifies core internal services are running. Default: True.
-            default_value: true
```

Comment:
Any reasons that we want to remove the default value here? I thought the field would always be set to true if the user does not explicitly set this field.
 Plus, removing a default value is likely be considered as a breaking change and we may need to wait until major release to implement it.

Reply:
Then what's gonna happen if user does not set this field explicitly after removing the default value from Terraform? Will this field be set to true, false, unset or it depends. As we currently set the default value as `True`, which means if user doesn't set this field, Terraform will set it as `True` and send it to the API. If we remove the default value from Terraform now and the API does not set it to `True`, which will possibly break the previous users  and it would be consider as a breaking change.

Reply:
What do you suggest then? The team agreed to not change the API behavior and leave it at False. I agree that it is a breaking change. The risk I see is that a Runtime would get deleted and recreated which, unless I am mistaken, is happening today if we don't change this because the default value does not match the API. If we were to go with the breaking change, when would be the next major release? If too far away, is there any way to merge this change beforehand.

Reply:
We don't have a clear plan on when the next major release is going to be, but it will likely not happen until next year. I'll have to discuss with my team to see if there's any ways we can work around this problem. 

A quick question, as I'm trying to understand the behavior. Do you know what value will be set to this field if users do not configure it? It should be either true or false, since it is a boolean, right? Looks like the API does not return the value, though.

Reply:
Nothing gets set if it is not actively provided. Our control plane will consider that it is false if the key is not there.

If set, it will add an optional attribute `report-system-health` to the underlying GCE instance. Otherwise that attribute won't exist.

If this is acceptable behavior, we would really need to find a solution before next year as we have customers expecting a Terraform update (and at this point, I think having some of their Runtime deleted would be better than not having the feature)

Reply:
Okay! Understand! Thanks for your explanation. Will working on finding possible solutions to implement the feature and avoid breaking changes!

Reply:
Just talked with the team, I think we can't change the default value until the next major release, but we can at least implement the update feature. I will run a couple of local tests later this afternoon and will leave my suggestion after. Sorry for the delay on this issue.